### PR TITLE
FIX(server): Stale user pointers in whisper cache 

### PR DESCRIPTION
**Pull request by Krzmbrzl**
Since the whisper cache entry was computed while holding the read-lock,
it was possible for a user in that cache to get deleted while the voice
thread drops the read lock in order to upgrade it to a write lock. In
such a case, the cache entry would contain an invalid (stale) user
pointer, which can lead to crashes when using the cache later on.

By moving the cache entry computation into the block in which we already
hold a read lock, we turn computing the cache and adding it to the cache
store into an atomic operation. As soon as the cache entry is in the
store, deleting a user is no longer an issue as that will implicitly
clear all whisper caches (thereby preventing stale cache entries).

Additionally, the cache creation has been factored out into its own function (for improved readability)

### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

File: src/murmur/Messages.cpp [hunk 1]
```diff
@@ -2199,7 +2199,7 @@ void Server::msgVoiceTarget(ServerUser *uSource, MumbleProto::VoiceTarget &msg)
 				unsigned int id = t.channel_id();
 				if (qhChannels.contains(id)) {
 					WhisperTarget::Channel wtc;
-					wtc.iId       = static_cast< int >(id);
+					wtc.iId       = id;
```
Comment (davidebeatrici) [inline src/murmur/Messages.cpp:2202]: Why is this not required anymore?

File: src/murmur/ServerUser.h [hunk 2]
```diff
@@ -52,7 +52,7 @@ struct BandwidthRecord {
 
 struct WhisperTarget {
 	struct Channel {
-		int iId;
+		unsigned int iId;
```
Comment (davidebeatrici) [inline src/murmur/ServerUser.h:55]: Can we do this in a separate commit?

Comment (Krzmbrzl) [reply src/murmur/ServerUser.h:55]: Sure we can, but why? To me this is also just a refactoring 👀

Comment (davidebeatrici) [reply src/murmur/ServerUser.h:55]: It's not related to the changes mentioned in this PR though.

Comment (Krzmbrzl) [reply src/murmur/ServerUser.h:55]: It's part of the whisper cache creation refactoring :shrug:

**Review by davidebeatrici — APPROVED** [review]

File: src/murmur/ServerUser.h (hunk 2 above)
Comment (Krzmbrzl) [reply src/murmur/ServerUser.h]: Given that the different signedness is not causing any bug (that I am aware of), I wouldn't call it a fix.

Anyway, I now have added a dedicated refactor commit that cleaned up the other members in the struct as well.

**Comment by Krzmbrzl** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|1.5.x|[](https://github.com/mumble-voip/mumble/pull/6382)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport) and see the [Github Action logs](https://github.com/mumble-voip/mumble/actions/runs/8582138444) for details

File: src/murmur/ServerUser.h (hunk 2 above)
Comment (davidebeatrici) [reply src/murmur/ServerUser.h]: Excellent.
